### PR TITLE
boards: mec: mec15modular: Update documentation with power management setup

### PR DESCRIPTION
**Pull request by albertofloyd**
Add details for features exposed via headers.
Add power management recommended setup.
Add setup information to generate binary that can be flashed into SPI chip.

Signed-off-by: Jose Alberto Meza <jose.a.meza.arellano@intel.com>

### CODE REVIEW

File: boards/arm/mec1501modular_assy6885/doc/index.rst [hunk 1]
```diff
@@ -22,23 +22,28 @@ Hardware
...
-- 2 Sockets for SPI NOR chips
-- One reset and VCC_PWRDGD pushbuttons
-- One external PCA9555 I/O port with jumper selectable I2C address.
-- One external LTC2489 delta-sigma ADC with jumper selectable I2C address.
-- Board power jumper selectable from +5V 2.1mm/5.5mm barrel connector or USB Micro A connector.
+- Independent Hardware Driven PS/2 Ports
+
+At difference from MEC15xx evaluation board, modular MEC1501 exposes the pins
+in 2 different ways:
+1) Standalone mode via headers
```
Comment (dbkinder) [inline boards/arm/mec1501modular_assy6885/doc/index.rst]: There needs to be a blank line before the first item in a list (or sublist), and the sublist must be indented, so change to:
```
At difference from the MEC15xx evaluation board, the modular MEC1501
exposes pins in two different ways:

1) Standalone mode via headers

   - GPIOs
   - PWM5
   - JTAG/SWD, ETM, and MCHP Trace ports
   - eSPI bus
   - SMB0

2) Mated mode with another platform that has a high-density MECC connector.

   - FAN0, PWM0, SMB0, SMB1, SMB4, and SMB5
   - eSPI bus
   - Breathing/Blinking LEDs
```

File: boards/arm/mec1501modular_assy6885/doc/index.rst [hunk 2]
```diff
@@ -22,23 +22,28 @@ Hardware
...
+- JTAG/SWD, ETM and MCHP Trace ports
+- eSPI bus
+- SMB0
+
+2) Mated mode with another platform that has a high density MECC connector.
+- FAN0, PWM0, SMB0, SMB1, SMB4 and SMB5
+- eSPI bus
+- Breathing/Blinking LEDs
+
+Similarly, to power the board +5V USB Micro A connector or from MECC connector.
```
Comment (dbkinder) [inline boards/arm/mec1501modular_assy6885/doc/index.rst]: I think this should read:
```
The board is powered through the +5V USB Micro A connector or from the MECC connector.
```

File: boards/arm/mec1501modular_assy6885/doc/index.rst [hunk 3]
```diff
@@ -100,8 +117,8 @@ board. Advanced users may deviate from this recommendation.
 Jumper setting for MEC1501 Modular Assy 6885 Rev A1p0
 =====================================================
 
-Power-related jumpers
----------------------
+Power-related jumpers.
```
Comment (dbkinder) [inline boards/arm/mec1501modular_assy6885/doc/index.rst]: Don't add a period at the end (none of the other titles end with a period)

**Review by dbkinder — CHANGES_REQUESTED** [review]
Thanks for the update.  A few recommended edits.

**Comment by albertofloyd** [comment]
@dbkinder check updated edits addressing comments

File: boards/arm/mec1501modular_assy6885/doc/index.rst [hunk 4]
```diff
@@ -22,23 +22,31 @@ Hardware
...
-- One external LTC2489 delta-sigma ADC with jumper selectable I2C address.
-- Board power jumper selectable from +5V 2.1mm/5.5mm barrel connector or USB Micro A connector.
+- Independent Hardware Driven PS/2 Ports
+
+At difference from MEC15xx evaluation board, modular MEC1501 exposes the pins
+in 2 different ways:
+
+1) Standalone mode via headers
+
+- GPIOs
```
Comment (dbkinder) [inline boards/arm/mec1501modular_assy6885/doc/index.rst]: Almost... you forgot to indent these list items (3 spaces) so they show up under the numbered list item (see the recommended edit in the original comment)

File: boards/arm/mec1501modular_assy6885/doc/index.rst [hunk 5]
```diff
@@ -22,23 +22,31 @@ Hardware
...
+- PWM5
+- JTAG/SWD, ETM and MCHP Trace ports
+- eSPI bus
+- SMB0
+
+2) Mated mode with another platform that has a high density MECC connector.
+
+- FAN0, PWM0, SMB0, SMB1, SMB4 and SMB5
```
Comment (dbkinder) [inline boards/arm/mec1501modular_assy6885/doc/index.rst]: Same here, these bullet points should be indented (3 spaces)

File: boards/arm/mec1501modular_assy6885/doc/index.rst [hunk 6]
```diff
@@ -158,6 +181,19 @@ This is the recommended setup.
 |             |     1      |  Use 3.3V Shared channel(R)|
 +-------------+------------+----------------------------+
 
+Power management.
```
Comment (dbkinder) [inline boards/arm/mec1501modular_assy6885/doc/index.rst]: Would you remove the period at the end of this heading (and the "Boot-ROM Straps." above and "Jumper location map." below (Headings shouldn't end with a period).

**Review by dbkinder — CHANGES_REQUESTED** [review]

File: boards/arm/mec1501modular_assy6885/doc/index.rst (hunk 6 above)
Comment (albertofloyd) [reply boards/arm/mec1501modular_assy6885/doc/index.rst]: Periods removed for all these headings

File: boards/arm/mec1501modular_assy6885/doc/index.rst (hunk 5 above)
Comment (albertofloyd) [reply boards/arm/mec1501modular_assy6885/doc/index.rst]: Corrected now

File: boards/arm/mec1501modular_assy6885/doc/index.rst (hunk 4 above)
Comment (albertofloyd) [reply boards/arm/mec1501modular_assy6885/doc/index.rst]: Done

**Review by dbkinder — APPROVED** [review]
+1 for doc changes, thanks.

**Pull request merged**